src/lib/core/overlay/overlay-ref.ts

```diff
@@ -101,18 +103,28 @@ export class OverlayRef implements PortalHost {
     let backdropToDetach = this._backdropElement;
 
     if (backdropToDetach) {
-      backdropToDetach.classList.remove('md-overlay-backdrop-showing');
-      backdropToDetach.classList.remove(this._state.backdropClass);
-      backdropToDetach.addEventListener('transitionend', () => {
-        backdropToDetach.parentNode.removeChild(backdropToDetach);
+      let onTransitionEnd = () => {
+        // It may not be attached to anything in certain cases (e.g. unit tests).
+        if (backdropToDetach && backdropToDetach.parentNode) {
+          backdropToDetach.parentNode.removeChild(backdropToDetach);
+        }
 
         // It is possible that a new portal has been attached to this overlay since we started
         // removing the backdrop. If that is the case, only clear the backdrop reference if it
         // is still the same instance that we started to remove.
         if (this._backdropElement == backdropToDetach) {
           this._backdropElement = null;
         }
-      });
+      };
+
+      backdropToDetach.classList.remove('md-overlay-backdrop-showing');
+      backdropToDetach.classList.remove(this._state.backdropClass);
```

Comment:
That's the custom class that can be added to the backdrop. It has been handled like that since https://github.com/angular/material2/commit/0b54668b4da909c4704135327b2d877446a7a295, I only moved it around.